frontend/catalyst/utils/tracing.py

```diff
@@ -16,45 +16,200 @@
 Tracing module.
```

Comment:
Let's rename this file and module docstring to be more descriptive of what it does and to separate from other tracing related modules :)
Maybe `contexts.py`?

Reply:
Done